dbms/src/Flash/Coprocessor/DAGContext.h

```diff
@@ -328,6 +328,8 @@ class DAGContext
     void setProcessListEntry(std::shared_ptr<ProcessListEntry> entry) { process_list_entry = entry; }
     std::shared_ptr<ProcessListEntry> getProcessListEntry() const { return process_list_entry; }
 
+    void addTableLock(const TableLockHolder & lock) { table_locks.push_back(lock); }
```

Comment:
yes, because `addTableLock` will only be called by `DAGStorageInterpreter` in a single thread, there is no data race, in line with `IBlockInputStream::addTableLock`.